### PR TITLE
Refactoring of pallet-lending

### PR DESCRIPTION
**Pull request by andor0**
## Issue
https://app.clickup.com/t/2c1ym7v

### CODE REVIEW

File: frame/lending/src/models/market_index.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+use frame_support::pallet_prelude::*;
+
+pub type MarketId = u32;
```
[the remark this answers is not in the document]
Comment (andor0) [reply frame/lending/src/models/market_index.rs:3]: newtype wrapping uses for type safety in function calls
type alias uses in RPC

[the remark this answers is not in the document]
Comment (benluelo) [reply frame/lending/src/models/market_index.rs:3]: We can't use the newtype in the RPC?

[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply frame/lending/src/models/market_index.rs:3]: not sure we need to have separate rs file. there not so many models. can we use models.rs?

File: frame/lending/src/lib.rs [hunk 2]
```diff
@@ -1131,12 +1138,16 @@ pub mod pallet {
...
 			ensure!(
-				config_input.updatable.collateral_factor > 1.into(),
-				Error::<T>::CollateralFactorMustBeMoreThanOne
+				config_input
+					.updatable
+					.collateral_factor
+					.try_into_validated::<MoreThanOne>()
+					.is_ok(),
+				Error::<T>::CollateralFactorIsNotValid
 			);
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply frame/lending/src/lib.rs:1151]: yes, it seems that check should go into validated impl, not here

**Comment by dzmitry-lahoda** [comment]
seems obsolete for now.

**Comment by dzmitry-lahoda** [comment]
will close you do not mind? :) so will not find it when searching for lending prs :)

**Pull request closed without merging**